src/corenetwork/lteip/LteIp.cc

```diff
@@ -132,195 +135,226 @@ void LteIp::endService(cPacket *msg)
+
     FlowControlInfo *controlInfo = new FlowControlInfo();
     controlInfo->setSrcAddr(src.getInt());
     controlInfo->setDstAddr(dest.getInt());
     controlInfo->setSrcPort(srcPort);
     controlInfo->setDstPort(dstPort);
     controlInfo->setSequenceNumber(seqNum_++);
-    controlInfo->setHeaderSize(headerSize);
+    controlInfo->setHeaderSize(headerSize.get());
     printControlInfo(controlInfo);
 
     datagram->setControlInfo(controlInfo);
 
     //** Send datagram to lte stack or LteIp peer **
     send(datagram, outputgate);
-    delete ipControlInfo;
 }
 
-void LteIp::toTransport(cPacket * msg)
-{
+void LteIp::toTransport(cPacket * msg) {
     // msg is an IP Datagram (from Lte stack or IP peer)
-    IPv4Datagram *datagram = check_and_cast<IPv4Datagram *>(msg);
-    int protocol = datagram->getTransportProtocol();
+    inet::Packet *datagram = check_and_cast<inet::Packet *>(msg);
+    // int protocol = datagram->getTransportProtocol();
 
-    int gateindex = 0;
-    try
-    {
-        gateindex = mapping_.getOutputGateForProtocol(protocol);
-    }
-    catch (cRuntimeError)
-    {
-        EV << "Protocol mapping failed with protocol number : " << protocol << endl;
-        EV << "Packet dropped" << endl;
-        delete msg;
-        numForwarded_--;
-        numDropped_++;
-        return;
-    }
 
-        // transport packet
-    cPacket *transportPacket = datagram->decapsulate();
+    // transport packet
+    //cPacket *transportPacket = datagram->decapsulate();
+    auto header = datagram->peekAtFront<Ipv4Header>();
 
     // create and fill in control info
-    IPv4ControlInfo *controlInfo = new IPv4ControlInfo();
-    controlInfo->setProtocol(datagram->getTransportProtocol());
-    controlInfo->setSrcAddr(datagram->getSrcAddress());
-    controlInfo->setDestAddr(datagram->getDestAddress());
-    controlInfo->setTypeOfService(datagram->getTypeOfService());
+    auto controlInfo = make_shared<Ipv4Header>();
+    controlInfo->setProtocol(header->getProtocol());
+    controlInfo->setSrcAddress(header->getSrcAddress());
+    controlInfo->setDestAddress(header->getDestAddress());
+    controlInfo->setTypeOfService(header->getTypeOfService());
+    auto packet = make_shared<inet::Packet>(*datagram);
 
     // XXX these two fields (interfaceId and datagram) are not actually of interest for us..
 
     // interfaceId should be the interface on which the datagram arrived,
     // or -1 if it was created locally
-    controlInfo->setInterfaceId(-1);
+    //controlInfo->setInterfaceId(-1);
     // original IP datagram might be needed in upper layers
-    controlInfo->setOrigDatagram(datagram);
+    //controlInfo->setOrigDatagram(datagram);
 
     // attach control info
-    transportPacket->setControlInfo(controlInfo);
+    //transportPacket->setControlInfo(controlInfo);
+    packet->popAtFront<Ipv4Header>();
+    packet->insertAtFront(Ptr<Ipv4Header>(controlInfo.get()));
```

Comment:
I guess this is a temporary solution.

Reply:
This was a left-over from the migration process and completely unnecesssary. We removed it completely.